src/Shared/FileMatcher.cs

```diff
@@ -547,16 +587,20 @@ private static void PreprocessFileSpecForSplitting
         /// Removes the leading ".\" from all of the paths in the array. 
         /// </summary>
         /// <param name="paths">Paths to remove .\ from.</param>
-        private static void RemoveInitialDotSlash
+        private static IEnumerable<string> RemoveInitialDotSlash
         (
-            string[] paths
+            IEnumerable<string> paths
         )
         {
-            for (int i = 0; i < paths.Length; i++)
+            foreach (string path in paths)
```

Comment:
This can throw.

Reply:
[Fixed](https://github.com/maca88/msbuild/blob/b5574289426c99e9567a43fe17d4731e1f4f5380/src/Shared/FileMatcher.cs#L321)